debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java

```diff
@@ -604,15 +598,73 @@ private static BigInteger resolveNextChangeFromScnRange(String nextChangeValue)
         return value.equals(MAX_SCN_BI) ? MAX_SCN_BI : value;
     }
 
-    private static void logLogMinerLogEntries(Connection connection) throws SQLException {
-        LOGGER.debug("Log entries registered with LogMiner are:");
-        try (PreparedStatement statement = connection.prepareStatement("SELECT * FROM V$LOGMNR_LOGS"); ResultSet rs = statement.executeQuery()) {
+    /**
+     * Helper method that will dump the state of various critical tables used by the LogMiner implementation
+     * to derive state about which logs are to be mined and processed by the Oracle LogMiner session.
+     *
+     * @param connection the database connection
+     */
+    private static void logDatabaseState(Connection connection) {
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Configured redo logs are:");
+            try {
+                logQueryResults(connection, "SELECT * FROM V$LOGFILE");
+            }
+            catch (SQLException e) {
+                LOGGER.debug("Failed to obtain redo log table entries", e);
+            }
+            LOGGER.debug("Available archive logs are:");
+            try {
+                logQueryResults(connection, "SELECT * FROM V$ARCHIVED_LOG");
+            }
+            catch (SQLException e) {
+                LOGGER.debug("Failed to obtain archive log table entries", e);
+            }
+            LOGGER.debug("Available logs are:");
+            try {
+                logQueryResults(connection, "SELECT * FROM V$LOG");
+            }
+            catch (SQLException e) {
+                LOGGER.debug("Failed to obtain log table entries", e);
+            }
+            LOGGER.debug("Log history last 24 hours:");
+            try {
+                logQueryResults(connection, "SELECT * FROM V$LOG_HISTORY WHERE FIRST_TIME >= SYSDATE - 1");
+            }
+            catch (SQLException e) {
+                LOGGER.debug("Failed to obtain log history", e);
+            }
+            LOGGER.debug("Log entries registered with LogMiner are:");
+            try {
+                logQueryResults(connection, "SELECT * FROM V$LOGMNR_LOGS");
+            }
+            catch (SQLException e) {
+                LOGGER.debug("Failed to obtain registered logs with LogMiner", e);
+            }
+        }
+    }
+
+    /**
+     * Helper method that dumps the result set of an arbitrary SQL query to the connector's logs.
+     *
+     * @param connection the database connection
+     * @param query the query to execute
+     * @throws SQLException thrown if an exception occurs performing a SQL operation
+     */
+    private static void logQueryResults(Connection connection, String query) throws SQLException {
+        try (PreparedStatement s = connection.prepareStatement(query); ResultSet rs = s.executeQuery()) {
```

Comment:
Would it help to use one of the utility methods on `JdbcConnection`?

Reply:
Adjusted the API to pass in `OracleConnection` so we could access those methods.  I'll add making sure we align to using that class primarily for connection operations as it relates to `LogMinerHelper` in [DBZ-2552](https://issues.redhat.com/browse/DBZ-2552).